search/src/test/scala/weco/api/search/json/CatalogueJsonUtilTest.scala

```diff
@@ -5,622 +5,128 @@ import io.circe.parser.parse
+  // at the top level, nested in an array, and nested in an object.
+  def nestedJsonWithField(fieldName: String): Json =
+    parse(
+      s"""
+        {
+          "$fieldName": ["V00123", "b12345"],
+          "name": "A picture of a cat",
+          "arrayField": [
+            {
+              "$fieldName": ["V00123-1"],
+              "name": "The first image"
+            },
+            {
+              "$fieldName": ["V00123-2"],
+              "name": "The second image"
+            }
+          ],
+          "objectField": {
+            "V00456": {
+              "$fieldName": ["V00456", "b56789"],
+              "name": "A picture of a dog"
+            }
+          }
+        }
+        """
+    ).right.get
+
+  def nestedJsonWithoutField: Json =
+    parse(
+      """
+        {
+          "name": "A picture of a cat",
+          "arrayField": [
+            {
+              "name": "The first image"
+            },
+            {
+              "name": "The second image"
+            }
+          ],
+          "objectField": {
+            "V00456": {
+              "name": "A picture of a dog"
+            }
+          }
+        }
+        """
+    ).right.get
+
+  def jsonWithField(fieldName: String): Json =
+    parse(
+      s"""
+        {
+          "$fieldName": ["V00123", "b12345"],
+          "name": "A picture of a fish"
+        }
+        """
+    ).right.get
+
+  def jsonWithoutField: Json =
+    parse(
+      """
+        {
+          "name": "A picture of a fish"
+        }
+        """
+    ).right.get
+
+  val testCases = Table(
+    ("fieldName", "workInclude"),
+    ("items", WorkInclude.Items),
+    ("holdings", WorkInclude.Holdings),
+    ("subjects", WorkInclude.Subjects),
+    ("genres", WorkInclude.Genres),
+    ("contributors", WorkInclude.Contributors),
+    ("production", WorkInclude.Production),
+    ("languages", WorkInclude.Languages),
+    ("notes", WorkInclude.Notes),
+    ("images", WorkInclude.Images),
+    ("parts", WorkInclude.Parts),
+    ("partOf", WorkInclude.PartOf),
+    ("precededBy", WorkInclude.PrecededBy),
+    ("succeededBy", WorkInclude.SucceededBy)
+  )
+
+  describe("WorkJsonOps") {
+    it("includes/omits identifiers based on the work include") {
+      nestedJsonWithField("identifiers").withIncludes(
+        WorksIncludes(WorkInclude.Identifiers)
+      ) shouldBe nestedJsonWithField("identifiers")
+
+      nestedJsonWithField("identifiers").withIncludes(WorksIncludes.none) shouldBe nestedJsonWithoutField
     }
 
-    describe("contributors") {
-      val w = indexedWork().contributors(
-        List(
-          Contributor(agent = Agent(randomAlphanumeric(25)), roles = Nil)
-        )
-      )
-
-      it("includes contributors with WorkInclude.Contributors") {
-        val includes = WorksIncludes(WorkInclude.Contributors)
-        val json = w.asJson(includes)
+    it("includes/omits fields based on the work include") {
```

Comment:
(Similar to my comment on another #430)
Because this is refactoring and replicates what was already there, these aren't change requests, but a bit of work to consider for the future.

This test guarantees that the given fields will be removed from the top level if they are present.  However, it does not show that they will not be removed from a nested position.

When reading this change in isolation, it took a while to realise that that is why `identifiers` has its own test, distinct from all the other properties.

Also for complete coverage, I'd like to see 

* ensure that it removes more than one field when excluded
* ensure that it retains more than one field when included

As it is, the test does cover the remaining condition - 
* ensure that it still succeeds if there is nothing to do.

Reply:
I’ve added a couple of extra test cases and some comments to explain why `identifiers` are special.